conan/test_package/conanfile.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from conans import ConanFile, CMake, tools, RunEnvironment
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        with tools.environment_append(RunEnvironment(self).vars):
```

Comment:
I have already seen this in another package. @uilianries it seems it could be nice to have it automated in some conan tool, to simplify the ``test()`` method

Reply:
thanks to @solvingj. I could merge into conan on a next PR